cmd/cronosd/main.go

```diff
@@ -1,14 +1,37 @@
 package main
 
 import (
+	"fmt"
 	"os"
+	"os/signal"
+	"syscall"
+	"time"
 
 	svrcmd "github.com/cosmos/cosmos-sdk/server/cmd"
 	"github.com/crypto-org-chain/cronos/v2/app"
 	"github.com/crypto-org-chain/cronos/v2/cmd/cronosd/cmd"
 )
 
+import "C"
+
 func main() {
+	// Create a channel to listen for incoming signals
+	sigChan := make(chan os.Signal, 1)
+
+	// Notify the channel for SIGINT and SIGTERM signals
+	signal.Notify(sigChan, syscall.SIGINT, syscall.SIGTERM)
+
+	// Start a goroutine to handle the signal
+	go func() {
+		sig := <-sigChan
```

Comment:
does it work, if main thread quit, it won't wait for the goroutine

Reply:
yes, I'll check further information

Reply:
In ethermint's `server/start.go`, there's some effort to make the quit graceful, but there are some known issues.

Reply:
if you want to try, you can port these changes: https://github.com/cosmos/cosmos-sdk/pull/16203

Reply:
there is 
```
ListenForQuitSignals(g, true, cancelFn, svrCtx.Logger)
```
which also syscall.SIGINT, syscall.SIGTERM, 
so this code, buy a time ListenForQuitSignals settles

Reply:
> so this code, buy a time ListenForQuitSignals settles

why, if main thread quit, the goroutine quit unconditionally, if main thread stays, we don't need the goroutine, right?

Reply:
What the new code does is like this right?
- if main thread can't finish graceful shutdown in 2 seconds, it forcefully shutdown with `os.Exit`
- if main thread shutdown preemptively, it can't do anything.

Reply:
with latest code, ctrl-c works.
I'll close this pr